### PR TITLE
Marathon leadership-aware sync

### PR DESCRIPTION
**Pull request by dankraw**
Scheduled Marathon-Consul sync will run:
1. only when specified node has leadership
2. always when forced

### CODE REVIEW

File: sync/sync_test.go [hunk 1]
```diff
@@ -169,16 +223,21 @@ func TestSync_WithMarathonProblems(t *testing.T) {
...
 	consul := errorConsul{}
-	sync := New(marathon, consul)
+	sync := newSync(marathon, consul)
 	// when
 	err := sync.SyncServices()
 	// then
 	assert.Error(t, err)
 }
 
+func newSync(marathon marathon.Marathoner, service consul.ConsulServices) *Sync {
```
Comment (janisz) [inline sync/sync_test.go]: I'm not sure if this name is good. What's difference between `newSync` and `New`? `newSync` just call `New` with default config, name should represent it. How about `NewWithDefaultConfig`?

File: README.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ make test
 - Provided HTTP healtcheck will be transferred to Consul.
 - Labels with `tag` value will be converted to Consul tags, `marathon` tag is added by default
  (e.g, `labels: ["public":"tag", "varnish":"tag", "env": "test"]` → `tags: ["public", "varnish", "marathon"]`).
+- To schedule Marathon-consul sync one of two options has to be set: `sync-leader` (run sync only on specified node [leadership](https://mesosphere.github.io/marathon/docs/rest-api.html#get-v2-leader)) or `sync-force` (run always).
```
Comment (janisz) [inline README.md]: IMO we should describe what's the difference between `sync-leader` and `marathon-location`. Also describe when sync will take place so I'll write something like this:
> (run sync only on node that is Marathon leader ([marthon leader] equals `sync-leader`)

File: config/config.go [hunk 3]
```diff
@@ -68,6 +68,8 @@ func (config *Config) parseFlags() {
 
 	// Sync
 	flag.DurationVar(&config.Sync.Interval, "sync-interval", 15*time.Minute, "Marathon-consul sync interval")
+	flag.BoolVar(&config.Sync.Force, "sync-force", false, "Force leadership-independent Marathon-consul sync")
```
Comment (janisz) [inline config/config.go]: Default should be `true` or `sync-leader` should has default `hostname:8080`

File: README.md [hunk 4]
```diff
@@ -73,9 +74,11 @@ marathon-protocol      | http                  | Marathon protocol (http or http
 marathon-username      |                       | Marathon username for basic auth
 metrics-interval       | 30s                   | Metrics reporting [interval](https://golang.org/pkg/time/#Duration) **Note:** While using file configuration intervals should be provided in *nanoseconds*
 metrics-location       |                       | Graphite URL (used when metrics-target is set to graphite)
-metrics-prefix         | default               | Metrics prefix (default is resolved to <hostname>.<app_name>
+metrics-prefix         | default               | Metrics prefix (resolved to `<hostname>.<app_name>` by default)
 metrics-target         | stdout                | Metrics destination stdout or graphite
 sync-interval          | 15m0s                 | Marathon-consul sync [interval](https://golang.org/pkg/time/#Duration) **Note:** While using file configuration intervals should be provided in *nanoseconds*
+sync-leader            |                       | Marathon instance for which [leadership](https://mesosphere.github.io/marathon/docs/rest-api.html#get-v2-leader) the Marathon-consul sync is enabled
```
Comment (janisz) [inline README.md]: We should write what exactly we are expecting here - Hostname, URL or URI?

File: sync/sync.go [hunk 5]
```diff
@@ -42,6 +53,9 @@ func (s *Sync) SyncServices() error {
 }
 
 func (s *Sync) syncServices() error {
+	if check, err := s.shouldPerformSync(); !check {
```
Comment (janisz) [inline sync/sync.go]: Probably it's out of scope of this PR but I've just noticed we are not handling this error and any error generated by [`SyncServices`](https://github.com/allegro/marathon-consul/pull/20/files#diff-0c2cc759dcd70bc815a12c3ee5ce3270R38)

File: sync/sync_test.go (hunk 1 above)
Comment (dankraw) [inline sync/sync_test.go]: fixed

File: README.md (hunk 2 above)
Comment (dankraw) [inline README.md]: fixed the readme

File: config/config.go (hunk 3 above)
Comment (dankraw) [inline config/config.go]: OK, I will implement the default hostname resolving but I will have to add `sync-enabled` option as well, as we loose the ability to disable the whole thing now :)

File: README.md (hunk 4 above)
Comment (dankraw) [inline README.md]: fixed

File: sync/sync.go (hunk 5 above)
Comment (dankraw) [inline sync/sync.go]: done

File: config/config.go (hunk 3 above)
Comment (dankraw) [inline config/config.go]: done

**Comment by dankraw** [comment]
@janisz please review again :)

**Comment by janisz** [comment]
:+1:

**Pull request merged**